go/app/main.go

```diff
@@ -20,22 +26,149 @@ type Response struct {
 	Message string `json:"message"`
 }
 
+type Item struct {
+	Id           int
+	Name         string `json:"name"`
+	Category     string `json:"category"`
+	Img_filename string `json:"img_filename"`
+}
+
+type ItemWrapper struct {
+	Items []Item `json:"items"`
+}
+
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
+func hashString(s string) string {
+	h := sha256.New()
+	h.Write([]byte(s))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func trimPath(s string) string {
+	imgFileName := filepath.Base(s)
+	img := strings.TrimSuffix(imgFileName, filepath.Ext(imgFileName))
+	return img
+}
+
+func readItemsFromFile() (ItemWrapper, error) {
+	data, err := os.ReadFile("items.json")
+	if err != nil {
+		log.Printf("Failed to unmarshal items.json: %v", err)
+		return ItemWrapper{}, err
```

Comment:
この場合は空のItemWrapperを返すのではなく`nil`を返します。本当にtemWrapperの空を返したい時と区別するためです。他の部分も同様に直していただけますか？
```suggestion
		return nil, err
```